test/collection_test.js

```diff
@@ -32,7 +32,7 @@ describe("Collection", () => {
 
   function testCollection(options={}) {
     events = new EventEmitter();
-    const opts = Object.assign({events}, options);
+    const opts = {...events, ...options};
```

Comment:
Here we are changing the behavior too, but probably breaking something.

It should be: ``{events, ...options}``